collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -755,9 +768,10 @@ protected void onActivityResult(int requestCode, int resultCode, final Intent in
                  * Android 1.6) we want to handle images the audio and video
                  */
 
-                showDialog(SAVING_IMAGE_DIALOG);
-                Runnable runnable = () -> saveChosenImage(intent.getData());
-                new Thread(runnable).start();
+                ProgressDialogFragment progressDialog = ProgressDialogFragment.newInstance(getString(R.string.please_wait));
```

Comment:
The same here you can just call show on a result of `ProgressDialogFragment.newInstance()` instead of creating an object you use only once.